renku/core/models/datasets.py

```diff
@@ -262,6 +262,10 @@ class DatasetFile(Entity, CreatorsMixin):
 
     url = jsonld.ib(default=None, context='schema:url', kw_only=True)
 
+    based_on = jsonld.ib(
+        default=None, context='schema:isBasedOn', kw_only=True
```

Comment:
is there a reason you chose `isBasedOn` rather than `sameAs`? I guess you are assuming the user could modify the imported file?

Reply:
Yes. That was one of the reasons. The other reason is that `sameAs` can only be a URL while `isBasedOn` can also be a `CreativeWork`.

Reply:
we will have stable URIs for all file and dataset objects soon - we might need to revisit this choice then.